### PR TITLE
Example PIVideoPlayer (Window & View)

### PR DESCRIPTION
**Pull request by Xodia**
Example as fullscreen mode video. (Window and the video view)
Modify accessibility of PIVideoView from the window class in case we want to adjust the frame/add constraints to it.

### CODE REVIEW

**Comment by lk23lekkladf2** [comment]
Would you mind rebasing this pull request? Thanks!

**Comment by Xodia** [comment]
@jlmendezbonini Done.

File: Example/PIVideoPlayer/PIViewController.m [hunk 1]
```diff
@@ -18,9 +24,127 @@ @implementation PIViewController
...
+                                                          relatedBy:NSLayoutRelationEqual
+                                                             toItem:to
+                                                          attribute:NSLayoutAttributeHeight
+                                                         multiplier:1
+                                                           constant:0]];
+}
+
+// Transition
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
+
```
Comment (lk23lekkladf2) [inline Example/PIVideoPlayer/PIViewController.m:105]: - [ ] Remove extra space

File: Example/PIVideoPlayer/PIViewController.m [hunk 2]
```diff
@@ -18,9 +24,127 @@ @implementation PIViewController
...
+    if (!self.videoView.shouldLoopVideo) {
+        UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"End" message:@"Video ended" preferredStyle:UIAlertControllerStyleAlert];
+        UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleCancel handler:nil];
+        [alert addAction:okAction];
+        [self presentViewController:alert animated:YES completion:nil];
+    }
+}
+
+#pragma mark - PIVideoPlayerWindow Delegate Methods
+
```
Comment (lk23lekkladf2) [inline Example/PIVideoPlayer/PIViewController.m:143]: - [ ] Remove extra new line

File: Example/PIVideoPlayer/PIViewController.m [hunk 3]
```diff
@@ -18,9 +24,127 @@ @implementation PIViewController
...
+{
+    [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+    [self.view needsUpdateConstraints];
+}
+
+#pragma mark - PIVideoView Delegate Methods
+
+- (void)videoViewDidLoadVideo:(PIVideoView *)view
+{
+    // YAY
```
Comment (lk23lekkladf2) [inline Example/PIVideoPlayer/PIViewController.m:115]: - [ ] Replace with `NSLog(@"%@", NSStringFromSelector(_cmd));`

File: Example/PIVideoPlayer/PIViewController.m [hunk 4]
```diff
@@ -18,9 +24,127 @@ @implementation PIViewController
...
+{
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error" message:@"Could not load video" preferredStyle:UIAlertControllerStyleAlert];
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleCancel handler:nil];
+    [alert addAction:okAction];
+    [self presentViewController:alert animated:YES completion:nil];
+}
+
+- (void)videoViewDidBeginPlayingVideo:(PIVideoView *)view
+{
+    // YAY 2
```
Comment (lk23lekkladf2) [inline Example/PIVideoPlayer/PIViewController.m:129]: - [ ] Replace with `NSLog(@"%@", NSStringFromSelector(_cmd));`

File: Example/PIVideoPlayer/PIViewController.m [hunk 5]
```diff
@@ -18,9 +24,127 @@ @implementation PIViewController
...
+        [self presentViewController:alert animated:YES completion:nil];
+    }
+}
+
+#pragma mark - PIVideoPlayerWindow Delegate Methods
+
+
+- (void)videoPlayerWindowDidFinishPlaying:(PIVideoPlayerWindow *)window
+{
+    // YAY 3
```
Comment (lk23lekkladf2) [inline Example/PIVideoPlayer/PIViewController.m:147]: - [ ] Replace with `NSLog(@"%@", NSStringFromSelector(_cmd));`

File: Example/Podfile [hunk 6]
```diff
@@ -1,11 +1,11 @@
 source 'https://github.com/CocoaPods/Specs.git'
 
 target 'PIVideoPlayer', :exclusive => true do
-  pod "PIVideoPlayer", :path => "../"
+  pod "PIVideoPlayer"
 end
 
 target 'Tests', :exclusive => true do
-  pod "PIVideoPlayer", :path => "../"
```
Comment (lk23lekkladf2) [inline Example/Podfile:8]: @Xodia why did you remove `path`? I believe this was used to be able to modify the pod while working with the example project. @Haud ?
